File: Data/Module/TinyBuilder/Script/tzw.lua
--add the search result
package.path = "Data/module/TinyBuilder/Script/?.lua;" .. package.path
hotfix = require("hotfix.hotfix")
g_include_module = {}
function include(moduleName)
	local r = require(moduleName)
	g_include_module[moduleName] = 1
	return r
end

function tzw_engine_reload()
	print("on tzw_engine_reload")
	for k, v in pairs(g_include_module) do
		-- package.loaded[k] = nil
		-- require(k)
		hotfix.hotfix_module(k)
		print("hot fix for"..k)
	end
end


function test_fuck(a,b,c,d,e)
	print("In Lua Print!!!!!!!",a,b,c,d,e)
end

Main = include("Main")
-- init
function tzw_engine_init()
	Main.onEngineInit()
end

-- input event
function tzw_engine_input_event(input_event)
	Main.onEngineInputEvent(input_event)
end

--ui update
function tzw_engine_ui_update(dt)
	Main.handleUIUpdate(dt)
end

--specified window draw
function tzw_game_draw_window(arg)
	return Main.onDrawWindow(arg)
end

File: Data/Module/TinyBuilder/Script/tzw.py
# -*- coding:utf-8 -*-
import CubeEngine
import Main
import Reloader
print("love from Cube Engine")
a = 3
print("aaaaa", a)
def tzw_engine_init():
	print("tzw_engine_init from Python")
	Main.onEngineInit()

def tzw_engine_ui_update(dt):
	Main.handleUIUpdate(dt)
	
def tzw_engine_input_event(eventInfo):
	Main.onEngineInputEvent(eventInfo)

def tzw_on_game_ready():
	print("on Game Ready")
	pass

def tzw_on_gm_command(commandStr):
	print("the GM command str is", commandStr)
	if commandStr == "r":
		print("reload module")
		Reloader.reloadScript()
		pass

#specified window draw
def tzw_game_draw_window(arg):

	return Main.onDrawWindow(arg)

def tzw_engine_reload():
	pass

def on_game_start():
	Main.on_game_start()